pkg/bearertoken/grpc_test.go

```diff
@@ -0,0 +1,297 @@
+				if test.expectedMD == nil {
+					require.False(t, ok) // There should be no metadata in this case
+				} else {
+					require.True(t, ok)
+					assert.Equal(t, test.expectedMD, md)
+				}
+				return nil, nil
+			}
+			_, err = streamInterceptor(test.ctx, &grpc.StreamDesc{}, nil, "method", streamInvoker)
+			if test.expectedErr == "" {
+				require.NoError(t, err)
+			} else {
+				require.Error(t, err)
+				assert.Contains(t, err.Error(), test.expectedErr)
+			}
+		})
+	}
+}
+
+func TestServerInterceptors(t *testing.T) {
+	tests := []struct {
+		name        string
+		ctx         context.Context
+		expectedErr string
+		wantToken   string
+	}{
+		{
+			name:        "no token in context",
+			ctx:         context.Background(),
+			expectedErr: "",
+			wantToken:   "",
+		},
+		{
+			name:        "token in context",
+			ctx:         ContextWithBearerToken(context.Background(), "test-token"),
+			expectedErr: "",
+			wantToken:   "test-token",
+		},
+		{
+			name: "multiple tokens in metadata",
+			ctx: metadata.NewIncomingContext(context.Background(), metadata.MD{
+				Key: []string{"token1", "token2"},
+			}),
+			expectedErr: "malformed token: multiple tokens found",
+			wantToken:   "",
+		},
+		{
+			name: "valid token in metadata",
+			ctx: metadata.NewIncomingContext(context.Background(), metadata.MD{
+				Key: []string{"valid-token"},
+			}),
+			expectedErr: "",
+			wantToken:   "valid-token",
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			// Test unary server interceptor
+			unaryInterceptor := NewUnaryServerInterceptor()
+			unaryHandler := func(ctx context.Context, _ any) (any, error) {
+				token, ok := GetBearerToken(ctx)
+				if test.wantToken == "" {
+					assert.False(t, ok, "expected no token")
+				} else {
+					assert.True(t, ok, "expected token to be present")
+					assert.Equal(t, test.wantToken, token)
+				}
+				return nil, nil
+			}
+
+			_, err := unaryInterceptor(test.ctx, nil, &grpc.UnaryServerInfo{}, unaryHandler)
+			if test.expectedErr == "" {
+				require.NoError(t, err)
+			} else {
+				require.Error(t, err)
+				assert.Contains(t, err.Error(), test.expectedErr)
+			}
+
+			// Test stream server interceptor
+			streamInterceptor := NewStreamServerInterceptor()
+			mockStream := &mockServerStream{ctx: test.ctx}
+			streamHandler := func(_ any, stream grpc.ServerStream) error {
+				token, ok := GetBearerToken(stream.Context())
+				if test.wantToken == "" {
+					assert.False(t, ok, "expected no token")
+				} else {
+					assert.True(t, ok, "expected token to be present")
+					assert.Equal(t, test.wantToken, token)
+				}
+				return nil
+			}
+
+			err = streamInterceptor(nil, mockStream, &grpc.StreamServerInfo{}, streamHandler)
+			if test.expectedErr == "" {
+				require.NoError(t, err)
+			} else {
+				require.Error(t, err)
+				assert.Contains(t, err.Error(), test.expectedErr)
+			}
+		})
+	}
+}
+
+func TestClientUnaryInterceptorWithBearerToken(t *testing.T) {
```

Comment:
`WithBearerToken` suffix is redundant in the test names, this is implied by the package name